src/ui/NoNameFacet/NoNameFacetHeader/NoNameFacetHeader.ts

```diff
@@ -0,0 +1,76 @@
+import { $$ } from '../../../utils/Dom';
+import { l } from '../../../strings/Strings';
+import { SVGIcons } from '../../../utils/SVGIcons';
+import { SVGDom } from '../../../utils/SVGDom';
+import { INoNameFacetOptions } from '../NoNameFacetOptions';
+import { NoNameFacetHeaderClear } from './NoNameFacetHeaderClear';
+import { NoNameFacetHeaderCollapseToggle } from './NoNameFacetHeaderCollapseToggle';
+import { NoNameFacetHeaderOperatorToggle } from './NoNameFacetHeaderOperatorToggle';
+
+export interface INoNameFacetHeaderOptions {
+  rootFacetOptions: INoNameFacetOptions;
+}
+
+export class NoNameFacetHeader {
+  public element: HTMLElement;
+  private waitAnimationElement: HTMLElement;
+  private clearButton: NoNameFacetHeaderClear;
+  private operatorToggle: NoNameFacetHeaderOperatorToggle;
+  private collapseToggle: NoNameFacetHeaderCollapseToggle;
+
+  constructor(private options: INoNameFacetHeaderOptions) {
+    this.element = $$('div', { className: 'coveo-facet-header' }).el;
+    $$(this.element).append(this.createTitleSection());
+    $$(this.element).append(this.createSettingsSection());
+  }
+
+  private createTitleSection() {
+    const section = $$('div', { className: 'coveo-facet-header-title-section' });
+
+    section.append(this.createTitle());
+
+    this.waitAnimationElement = this.createWaitAnimation();
```

Comment:
FYI, I think there's a recommendation done by UX that the animation should be tweaked a bit to only play after a significant delay has passed.

As in: if the query took more than 2 seconds, play the anim.